src/VisualStudio/Core/Def/RoslynActivityLogger.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    /// <summary>
+    /// Let people to inject <see cref="TraceSource"/> to monitor Roslyn activity
+    /// 
+    /// Here, we don't technically use TraceSource as it is meant to be used. but just as an easy 
+    /// way to log data to listeners.
+    /// 
+    /// this also involves creating string, boxing and etc. so, perf wise, it will impact VS quite a bit.
+    /// this also won't collect trace from Roslyn OOP for now. only in proc activity
+    /// </summary>
+    internal static class RoslynActivityLogger
```

Comment:
Is the plan to find this and call `SetLogger` via reflection from the extension?

Reply:
yep. I already have corresponding change in project system tool - https://github.com/dotnet/project-system-tools/pull/44